mixins/base-child.js

```diff
@@ -38,6 +38,25 @@ const baseChild = {
   },
 
   /**
+   * Creates a new metafield on a record.
+   * This also works to update the existing metafield
+   *
+   * @param {Number} parentId Parent record ID
+   * @param {Object} params Record properties
+   * @return {Promise} Promise that resolves with the result
+   * @public
+   */
+  createMetafield(parentId, params) {
+    // rearrange and override arguments
+    this.parentName = this.name;
+    this.name = 'metafield';
```

Comment:
You are changing "private" properties of the instance with these assignments.
What happens if you try to use say `shopify.productVariant.get(parentId)` after calling `shopify.productVariant.createMetafield(parentId, params)`?